target_setting_runs/jax_nesterov.py

```diff
@@ -0,0 +1,207 @@
+"""Submission file for a SGD with Nesterov optimizer in Jax."""
+
+import functools
+import inspect
+from typing import Callable, Iterable, Union
+
+from flax import jax_utils
+import jax
+import jax.numpy as jnp
+import optax
+
+from algorithmic_efficiency import spec
+
+
+def init_optimizer_state(workload: spec.Workload,
+                         model_params: spec.ParameterContainer,
+                         model_state: spec.ModelAuxiliaryState,
+                         hyperparameters: spec.Hyperparameters,
+                         rng: spec.RandomState) -> spec.OptimizerState:
+  """Creates a Nesterov optimizer and a learning rate schedule."""
+  del model_params
+  del model_state
+  del rng
+
+  # Create optimizer.
+  params_zeros_like = jax.tree_map(lambda s: jnp.zeros(s.shape_tuple),
+                                   workload.param_shapes)
+  opt_init_fn, opt_update_fn = static_inject_hyperparams(sgd)(
```

Comment:
Ah nice, I used it because as you say `if t < warmup_steps:` cannot be jitted, but just using `optax.join_schedules` is much better. I have also noticed that the optax implementation of the linear warmup (`optax.linear_schedule`) and the polynomial decay (`optax.polynomial_schedule`) seem to be equivalent to the init2winit implementation at first glance, so I'm using them now which requires less code. Or is there a difference I have missed? Not sure why they were re-implemented in init2winit.

Reply:
nope they should be the same, we kept our own because we both implemented them in parallel.